### PR TITLE
Add `options` field to UC catalog resource to support foreign catalog creation

### PR DESCRIPTION
**Pull request by andrewli81**
## Changes
Add `options` field to UC catalog resource to support foreign catalog creation. 
When creating a foreign catalog, on top of providing a connection name, user also needs to provide `options` field which gives the mapping between the foreign database and the catalog. 

https://docs.databricks.com/api/workspace/catalogs/create

## Tests

- [X] `make test` run locally
- [X] relevant change in `docs/` folder
- [ ] covered with integration tests in `internal/acceptance`
- [ ] relevant acceptance tests are passing
- [ ] using Go SDK

### CODE REVIEW

File: catalog/resource_catalog_test.go [hunk 1]
```diff
@@ -343,6 +343,63 @@ func TestCatalogCreateDeltaSharing(t *testing.T) {
...
+					Owner:          "f",
+				},
+			},
+		},
+		Resource: ResourceCatalog(),
+		Create:   true,
+		HCL: `
+		name = "a"
+		comment = "b"
+		options = {
```
Comment (andrewli81) [inline catalog/resource_catalog_test.go:395]: @tanmay-db I am rather unfamiliar with either Go or terraform, what I could not understand is the test result doesn't include the `options` field but I can't figure out why it's the case.

File: catalog/resource_catalog_test.go [hunk 2]
```diff
@@ -343,6 +343,68 @@ func TestCatalogCreateDeltaSharing(t *testing.T) {
...
+func TestCatalogCreateForeign(t *testing.T) {
+	d, err := qa.ResourceFixture{
+		Fixtures: []qa.HTTPFixture{
+			{
+				Method:   "POST",
+				Resource: "/api/2.1/unity-catalog/catalogs",
+				ExpectedRequest: catalog.CatalogInfo{
+					Name:    "foreign_catalog",
+					Comment: "b",
+					Options: map[string]string{
```
Comment (andrewli81) [inline catalog/resource_catalog_test.go:355]: Somehow, the request doesn't match the expected request here and does not include the "options" field, I am not sure if I am missing anything - but there doesn't seem to be anything extra needed to configure this.

Comment (andrewli81) [reply catalog/resource_catalog_test.go:355]:
```
=== FAIL: catalog TestCatalogCreateForeign (0.14s)
    testing.go:427: 
        	Error Trace:	/home/runner/work/terraform-provider-databricks/terraform-provider-databricks/qa/testing.go:427
        	            				/opt/hostedtoolcache/go/1.19.12/x64/src/net/http/server.go:2109
        	            				/opt/hostedtoolcache/go/1.19.12/x64/src/net/http/server.go:2947
        	            				/opt/hostedtoolcache/go/1.19.12/x64/src/net/http/server.go:1991
        	            				/opt/hostedtoolcache/go/1.19.12/x64/src/runtime/asm_amd64.s:1594
        	Error:      	Not equal: 
        	            	expected: map[string]interface {}{"comment":"b", "connection_name":"foo", "name":"a", "options":map[string]interface {}{"a":"b"}}
        	            	actual  : map[string]interface {}{"comment":"b", "connection_name":"foo", "name":"a"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,8 +1,5 @@
        	            	-(map[string]interface {}) (len=4) {
        	            	+(map[string]interface {}) (len=3) {
        	            	  (string) (len=7) "comment": (string) (len=1) "b",
        	            	  (string) (len=15) "connection_name": (string) (len=3) "foo",
        	            	- (string) (len=4) "name": (string) (len=1) "a",
        	            	- (string) (len=7) "options": (map[string]interface {}) (len=1) {
        	            	-  (string) (len=1) "a": (string) (len=1) "b"
        	            	- }
        	            	+ (string) (len=4) "name": (string) (len=1) "a"
        	            	 }
        	Test:       	TestCatalogCreateForeign
        	Messages:   	json strings do not match
```

Comment (andrewli81) [reply catalog/resource_catalog_test.go:355]: cc @nkvuong @tanmay-db would appreciate some help here.

Comment (nkvuong) [reply catalog/resource_catalog_test.go:355]: the OpenAPI spec has been updated, but we need a new release of Go SDK to incorporate those changes - should have one shortly

Comment (tanmay-db) [reply catalog/resource_catalog_test.go:355]: The newest version of Go SDK was released 2 days back and contains options field for Create and Update catalog.
image

https://github.com/databricks/databricks-sdk-go/releases/tag/v0.18.0

Were these the changes that the PR is waiting for?

Comment (tanmay-db) [reply catalog/resource_catalog_test.go:355]: I see, there are some issues with updating go sdk to latest version in terraform for now. The PR would need to be on hold until they are resolved. We are working on them.

**Review by tanmay-db — APPROVED** [review]
LGTM

**Pull request merged**